### DISCUSSION TITLE
FastAPI websockets Deployment on GCP Cloud Run

### DISCUSSION THREAD

**Discussion by HasnainKhanNiazi** [fastapi/fastapi#12049] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import WebSocket

class WebSocketManager:
    """
    A class to manage WebSocket connections.

    Attributes:
        websocket (WebSocket): The WebSocket connection.
    """
    def __init__(self):
        self.websocket = None

    async def connect(self, websocket: WebSocket):
        """
        Establish the WebSocket connection.

        Args:
            websocket (WebSocket): The WebSocket to connect.
        """
        await websocket.accept()
        self.websocket = websocket

    async def disconnect(self):
        """
        Disconnect the WebSocket if connected.
        """
        if self.websocket:
            try:
                await self.websocket.close()
            except RuntimeError:
                pass  # Ignore if websocket is already closed
        self.websocket = None

    async def send_answer(self, answer: str):
        """
        Send an answer to the connected WebSocket.

        Args:
            answer (str): The answer to send.
        """
        if self.websocket:
            await self.websocket.send_text(answer)

    async def receive_answer(self):
        """
        Receive an answer from the connected WebSocket.

        Returns:
            str: The received answer or None if not connected.
        """
        if self.websocket:
            answer = await self.websocket.receive_text()
            return answer
        return None

ws_manager = WebSocketManager()

@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket, app_id: str = Query(...)):
    """
    Handle WebSocket connections for real-time interactions.

    Args:
        websocket (WebSocket): The WebSocket connection.
    """
    await ws_manager.connect(websocket)
    while True:
        try:
            user_query = await websocket.receive_text()
            log_data("User query", user_query)
            start_time = time.time()
            .... rest of the code
        except Exception as e:
            log_error(e)
            await ws_manager.disconnect()
            break
```

### Description

I have deployed FastAPI websockets on my local system and it is working fine but I am having problems when I deploy it on GCP cloud run. But I am getting this error: `CloseCode.ABNORMAL_CLOSURE: 1006>` I tried to find a solution but I couldn't find anything.

### Operating System

Linux

### FastAPI Version

0.112.0

### Pydantic Version

2.8.2

### Python Version

3.11.9

**Comment by xarauzo** [comment]
Did you manage to solve this?

**Comment by lucasgomide** [comment, marked as answer]
It looks a Cloud Run issue. Have u read [it](https://cloud.google.com/run/docs/triggering/websockets)?
